### DISCUSSION TITLE
Quick Reblog: option to bump queued posts to the front

### DISCUSSION THREAD

**Discussion by captainpensive** [AprilSylph/XKit-Rewritten#724] [Suggestion Box]
I think it would be really useful to have an option whenever you queue a new post, that lets you choose if you want it at the end or the front of the queue. Would save you a lot of time not having to go to the queue tab and then slowly load through a dozen pages to get to the end and then finally bump it to the front.

If you're about to ask why you can't just reblog the post like usual, a lot of blogs prefer to space out their content as not to spam their followers, and being able to put a post at the front of your queue assures that without having to spend a lot of time manually sorting your queue, especially when your queue has 100s of posts.

**Comment by marcustyphoon** [comment]
Interesting idea! Just checked the internal API and this looks doable.

If this were to be an option for the real Tumblr post form, the hardest part would be figuring out what the post ID is in order to move it to the front, since when the post editor submit button is pressed it of course hasn't been generated yet. If this were just a Quick Reblog option, that's not an issue since we make the post ourselves.

Also, submit this as a feature suggestion to https://tumblr.com/support! Maybe lots of people want it, and it can become a native site feature.

`/api/v2/blog/[blogname]/posts/queue/reorder`, `{ post_id: "[postid]", insert_after: 0 }`
